File: 1078-remove-outermost-parentheses/1078-remove-outermost-parentheses.py
class Solution:
    def removeOuterParentheses(self, s: str) -> str:
        ans = ""
        st = []

        for i in range(len(s)):
            if s[i]=='(':
                if st:
                    ans += '('
                st.append(s[i])
            elif s[i] == ')':
                st.pop()
                if len(st) >= 1:
                    ans += ')'
        return ans


    

        

File: 1078-remove-outermost-parentheses/1078-remove-outermost-parentheses.java
class Solution {
    public String removeOuterParentheses(String s) {
        Stack<Character> st = new Stack<>();
        String ans = "";

        for(int i = 0 ; i < s.length(); i++){
            char c = s.charAt(i);

            if(c == '('){
                if(!st.isEmpty()) 
                ans += c;
                st.push(c);
            }
            else{
                st.pop();
                if(!st.isEmpty()) ans += ')';
            }
        }
        return ans;
    }
}